src/ppl/src/IOPlacer.cpp

```diff
@@ -600,8 +604,8 @@ bool IOPlacer::checkBlocked(Edge edge, int pos, int layer)
     if (blocked_interval.getLayer() == -1
         || blocked_interval.getLayer() == layer) {
       if (blocked_interval.getEdge() == edge
-          && pos >= blocked_interval.getBegin()
-          && pos <= blocked_interval.getEnd()) {
+          && pos > blocked_interval.getBegin()
+          && pos < blocked_interval.getEnd()) {
```

Comment:
how does this relate to writing the tcl?

Reply:
Not directly to writing the tcl, but when reading it. The pin placement when loading the tcl file differed from the tool run result. It was happening because of this check. When placing an individual pin, the tool creates a blocked region around it, and this check changes the position of the pins placed near the previous one. With this update, it allows touching the blocked area.